plugin.rb

```diff
@@ -591,10 +591,47 @@ def self.skip_db?
   if defined?(DiscourseAssign)
     on(:accepted_solution) do |post|
       next if SiteSetting.assignment_status_on_solve.blank?
-      Assigner.new(post.topic, post.acting_user).assign(
-        post.acting_user,
-        status: SiteSetting.assignment_status_on_solve,
-      )
+      assignements = Assignment.where(topic: post.topic)
```

Comment:
assignements -> assignments ;)

Reply:
You can do 
```ruby
assignments = Assignment.includes(:target).where(topic: post.topic)
```
which will preload the `target` to avoid an N+1 query thennnnn we avoid the whole switch statement and do 

```ruby
Assigner.new(assignment.target, assigned_user).assign(
          assigned_user,
          status: SiteSetting.assignment_status_on_solve,
        )
```